mllib/src/main/scala/org/apache/spark/ml/classification/ProbabilisticClassifier.scala

```diff
@@ -200,22 +200,17 @@ abstract class ProbabilisticClassificationModel[
     if (!isDefined(thresholds)) {
       probability.argmax
     } else {
-      val thresholds: Array[Double] = getThresholds
-      val probabilities = probability.toArray
+      val thresholds = getThresholds
       var argMax = 0
       var max = Double.NegativeInfinity
       var i = 0
       val probabilitySize = probability.size
       while (i < probabilitySize) {
-        if (thresholds(i) == 0.0) {
-          max = Double.PositiveInfinity
+        // thresholds are all > 0, excepting that at most one may be 0
+        val scaled = probability(i) / thresholds(i)
+        if (scaled > max) {
```

Comment:
Yeah, that occurred to me. It will never be selected because NaN isn't bigger than anything, including NegativeInfinity. If for some reason you have one class only (is this even valid?) you'd select this class, which is I guess correct.

Very small but positive prob / threshold? that should still work fine to the limits of machine precision here.